domain/network/network.py

```diff
@@ -0,0 +1,200 @@
+from dataclasses import asdict, dataclass
+from datetime import datetime, timedelta
+from typing import List, Set
+
+from dacite import from_dict
+
+from domain.network.node import Node, NodeState, Peer
+
+
+def uptime_to_up_since(uptime: float) -> int:
+    """ Calculate time when server went up considering server uptime. Remove precision miliseconds to avoid differences
+        between two instances with the same data calculated in different milliseconds
+
+    :param uptime: Time for how long the server is up in seconds
+    :type uptime: float
+    :return: Calculated datetime when server went up in second (rounded down)
+    :rtype: int
+    """
+    up_since_date = datetime.now() - timedelta(seconds=uptime)
+    return int(datetime.timestamp(up_since_date))
+
+
+@dataclass
+class AggregatedPeer:
+    """Aggregated data of a Peer retrieved by requesting a Node
+
+    :param id: id of peer
+    :type id: str
+
+    :param app_version: version of hathor-core running on peer
+    :type app_version: str
+
+    :param up_since: timestamp of when the peer went up
+    :type up_since: int
+
+    :param address: ip address of peer
+    :type address: str
+
+    :param state: state of peer
+    :type state: :py:class:`domain.network.node.NodeState`
+
+    :param last_message: timestamp of lat message
+    :type last_message: int
+
+    :param latest_timestamp: latest timestamp
+    :type latest_timestamp: int
+
+    :param sync_timestamp: timestamp of last sychronized block
+    :type sync_timestamp: int
+
+    :param warning_flags: list of warning flags, if any
+    :type warning_flags: List[str]
+
+    :param entrypoints: list entrypoints, if any
+    :type entrypoints: List[str]
+
+    :param connected_to: list of ids of nodes that peer is conected to
+    :type connected_to: Set[str]
+
+    """
+    id: str
+    app_version: str
+    up_since: int
```

Comment:
Why store `up_since` instead of `uptime`?

Reply:
`uptime` keep changing every second. `up_since` is the timestamp of when the node started (`now - uptime`) so it wont change for the same running node.